crates/oxc_linter/src/rules/eslint/no_unexpected_multiline.rs

```diff
@@ -0,0 +1,369 @@
+                    generics.span.end
+                } else {
+                    tagged_template_expr.tag.span().end
+                };
+                let span = Span::new(start, tagged_template_expr.span.end);
+                let src = ctx.source_range(span).as_bytes();
+                let Some(backtick) = memchr(b'`', src) else {
+                    return;
+                };
+                let Some(newline) = memchr(b'\n', src) else {
+                    return;
+                };
+                if newline < backtick {
+                    let backtick_span = Span::new(
+                        span.start + u32::try_from(backtick).unwrap(),
+                        span.start + u32::try_from(backtick + 1).unwrap(),
+                    );
+                    let diagnostic = OxcDiagnostic::warn(
+                        "Unexpected newline between template tag and template literal",
+                    )
+                    .with_label(backtick_span.label(
+                        "this is parsed as a tagged template, which may be unintentional",
+                    ))
+                    .with_help("If you did not intend for this to be a tagged template, insert ';' before the backtick");
+                    ctx.diagnostic_with_dangerous_fix(diagnostic, |fixer| {
+                        fixer.insert_text_before_range(backtick_span, ";")
+                    });
+                }
+            }
+            AstKind::BinaryExpression(binary_expr) => {
+                if binary_expr.operator != BinaryOperator::Division {
+                    return;
+                }
+                let Some(AstKind::BinaryExpression(parent_binary_expr)) =
+                    ctx.nodes().parent_kind(node.id())
+                else {
+                    return;
+                };
+                if parent_binary_expr.operator != BinaryOperator::Division {
+                    return;
+                }
+                let span = Span::new(binary_expr.left.span().end, parent_binary_expr.span().end);
+                let src = ctx.source_range(span);
+
+                let Some(newline) = memchr(b'\n', src.as_bytes()) else {
+                    return;
+                };
+                let Some(first_slash) = memchr(b'/', src.as_bytes()) else {
+                    return;
+                };
+                let Some(second_slash) = memrchr(b'/', src.as_bytes()) else {
+                    return;
+                };
+                if first_slash == second_slash {
+                    return;
+                }
+
+                // find index of first non-identifier character after the second slash
+                let ident_name: &str = if let Some(offset) = src[(second_slash + 1)..]
+                    .as_bytes()
+                    .iter()
+                    .position(|&b| !b.is_ascii_alphanumeric())
+                {
+                    &src[second_slash + 1..second_slash + 1 + offset]
```

Comment:
issue: I've seen ranged index on `source_text` cause rules to panic in the past (e.g. weird unicode characters cause this to end on a character boundary). Is it possible to refactor this logic to use `chars()` instead?

Reply:
good idea, I wish we had a good way to quickly tokenize some text here, but I refactored this to at least do a good enough job at finding identifier-like strings